hugolib/hugo_modules_test.go

```diff
@@ -1178,3 +1178,28 @@ target = "content/resources-b"
 	b.AssertFileContent("public/resources-a/subdir/about/index.html", "Single")
 	b.AssertFileContent("public/resources-b/subdir/about/index.html", "Single")
 }
+
+func TestMountData(t *testing.T) {
+	config := `baseURL = 'https://example.org/'
+languageCode = 'en-us'
+
+disableKinds = ["taxonomy", "term", "RSS", "sitemap", "robotsTXT", "page", "section"]
+
+[[module.mounts]]
+source = "data"
+target = "data"
+
+[[module.mounts]]
+source = "extra-data"
+target = "data/extra"
+`
+	b := newTestSitesBuilder(t).WithConfigFile("toml", config)
+
+	b.WithSourceFile("extra-data/test.yaml", "message: Hugo Rocks")
+
+	b.WithTemplatesAdded("index.html", "{{ site.Data.extra.test.message }}")
+
+	b.Build(BuildCfg{})
+
+	b.AssertFileContent("public/index.html", "Hugo Rocks")
```

Comment:
OK, the bug fix itself looks good, but I would appreciate if you could rewrite the test to use `NewIntegrationTestBuilder()`. The assertion API is similar. I'm slowly rewriting existing tests to use that setup, so I would hate to merge new stuff with the old testSitesBuilder that I'm trying to get rid of ...

Reply:
OK; done